pkg/plugin/prometheus/setup.go

```diff
@@ -0,0 +1,61 @@
+package prometheus
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/hellofresh/janus/pkg/errors"
+	"github.com/hellofresh/janus/pkg/plugin"
+	"github.com/hellofresh/janus/pkg/proxy"
+	"github.com/hellofresh/janus/pkg/router"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	adminRouter router.Router
+)
+
+func init() {
+	plugin.RegisterEventHook(plugin.StartupEvent, onStartup)
+	plugin.RegisterEventHook(plugin.AdminAPIStartupEvent, onAdminAPIStartup)
+
+	plugin.RegisterPlugin("prometheus", plugin.Plugin{
+		Action: setupPrometheus,
+	})
+}
+
+func onStartup(event interface{}) error {
+	e, ok := event.(plugin.OnStartup)
+	if !ok {
+		return fmt.Errorf("Could not convert event to startup type")
+	}
+	if adminRouter == nil {
+		return errors.New(http.StatusNotFound, "invalid admin router given")
+	}
+	if e.Config.Prometheus.Enabled {
+		adminRouter.GET(e.Config.Prometheus.MetricPath, prometheus.Handler().ServeHTTP)
```

Comment:
What do you think about moving this to `onAdminAPIStartup`? Since `onStartup` and `onAdminAPIStartup` happens concurrently it's hard to guarantee that `adminRouter` will be ready at this point. 
It's more likely that `onStartup` runs before `onAdminAPIStartup`.

Reply:
my issue with that is i couldn't access the config on `onAdminAPIStartup`

Reply:
True that event doesn't hold the global configuration. I'll work on an improvement to add that to it in the future. But for now, if you catch the configuration `onStartup` you can use it there later. THey are concurrent but `onStartup` starts way before the admin event so very likely it will be ready by then.